lib/ruby_indexer/lib/ruby_indexer/index.rb

```diff
@@ -21,9 +21,9 @@ class << self
       # Returns the real nesting of a constant name taking into account top level
       # references that may be included anywhere in the name or nesting where that
       # constant was found
-      sig { params(stack: T::Array[String], name: String).returns(T::Array[String]) }
+      sig { params(stack: T::Array[String], name: T.nilable(String)).returns(T::Array[String]) }
       def actual_nesting(stack, name)
-        nesting = stack + [name]
+        nesting = name ? stack + [name] : stack
```

Comment:
Why do we need these indexer changes? Should we add index tests for them too?
If it's to fix an indexer issue we discovered for this feature, let's open a separate PR for it.